### PR TITLE
refactor modules/packages structure

### PR DESCRIPTION
**Pull request by lightspeed69**

### CODE REVIEW

**Comment by lightspeed69** [comment]
1. triangle exception
2. vector is child class for linesegment
3. add test for Point Triangle exceptions

**Comment by lightspeed69** [comment]
Point 1, 3 are done. Point 2 is cancelled

File: line/line_segment.py [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
 
 
 class LineSegment:
     def __init__(self, a: Point, b: Point):
         self.a = a
         self.b = b
+        self._validate_equal_points()
+
+    def __repr__(self) -> str:
+        return f"A({self.a.x},{self.b.y}), B({self.b.x},{self.b.y})"
```
Comment (valentynmalyi) [inline line/line_segment.py:14]: Here you can use f"{self.a}"

File: test/test_triangle/test_triangle.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+from point import Point
+from triangle.triangle import exceptions, RightTriangle, Triangle
+
+
+class TestTriangle:
+    def test_validate_is_line(self):
+        with pytest.raises(exceptions.IsLine):
+            for a, b, c in permutations([Point(0, 0), Point(0, 0), Point(0, 1)], 3):
+                Triangle(a, b, c)
```
Comment (valentynmalyi) [inline test/test_triangle/test_triangle.py:13]: You don't call _validate_is_line method

File: test/test_triangle/test_triangle.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    def test_validate_is_point(self):
+        with pytest.raises(exceptions.IsPoint):
+            RightTriangle(Point(0, 0), Point(0, 0), Point(0, 0))
```
Comment (valentynmalyi) [inline test/test_triangle/test_triangle.py:17]: _validate_is_point

File: test/test_triangle/test_triangle.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class TestRightTriangle:
+    def test_validate_is_right(self):
+        assert isinstance(RightTriangle(Point(0, 0), Point(1, 0), Point(0, 2)), Triangle)
+        with pytest.raises(exceptions.IsNotRight):
+            RightTriangle(Point(3, 3), Point(1, 0), Point(0, 2))
```
Comment (valentynmalyi) [inline test/test_triangle/test_triangle.py:24]: _validate_is_right

File: triangle/triangle.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    def __repr__(self) -> str:
+        return f"A{self.a}, B{self.b}, C{self.c}"
+
+    def _validate(self) -> None:
+        self._validate_is_point()
+        self._validate_is_line()
+
+    def _validate_is_line(self) -> None:
+        for a, b, c in permutations([self.a, self.b, self.c], 3):
+            self._validate_line_segment(a, b, c)
```
Comment (valentynmalyi) [inline triangle/triangle.py:26]: I think this isn't enough
(0,0) (0,1) (0,2) isn't a triangle

File: triangle/triangle.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+
+class RightTriangle(Triangle):
+    def __init__(self, a: Point, b: Point, c: Point):
+        super().__init__(a, b, c)
+        self._validate_is_right(a, b, c)
+
+    def _validate_is_right(self, a: Point, b: Point, c: Point) -> None:
+        side = sorted([LineSegment(a, b).length, LineSegment(b, c).length, LineSegment(a, c).length])
+        if not isclose(side[2] ** 2, side[1] ** 2 + side[0] ** 2):
```
Comment (valentynmalyi) [inline triangle/triangle.py:47]: Here you can use square_lenght

File: triangle/triangle.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            raise exceptions.IsPoint(self.a)
+
+
+class RightTriangle(Triangle):
+    def __init__(self, a: Point, b: Point, c: Point):
+        super().__init__(a, b, c)
+        self._validate_is_right(a, b, c)
+
+    def _validate_is_right(self, a: Point, b: Point, c: Point) -> None:
+        side = sorted([LineSegment(a, b).length, LineSegment(b, c).length, LineSegment(a, c).length])
```
Comment (valentynmalyi) [inline triangle/triangle.py:46]: x,y,z = [1,2,3]

**Review by valentynmalyi — CHANGES_REQUESTED** [review]

File: triangle/triangle.py (hunk 7 above)
Comment (valentynmalyi) [reply triangle/triangle.py:46]: add TestPoint:test__eq__

File: line/line_segment.py (hunk 1 above)
Comment (lightspeed69) [reply line/line_segment.py]: done

File: triangle/triangle.py (hunk 7 above)
Comment (lightspeed69) [reply triangle/triangle.py]: done

File: triangle/triangle.py (hunk 5 above)
Comment (lightspeed69) [reply triangle/triangle.py]: done

**Review by valentynmalyi — APPROVED** [review]

**Pull request merged**